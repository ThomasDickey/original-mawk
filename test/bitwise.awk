#
# tests of bit-wise Arithmetic
#
BEGIN {
  x=0x11
  y=0x03
  printf "0x%X, 0x%X, 0x%X\n", and(x,y), or(x,y), xor(x,y)
  printf "0x%X, 0x%X\n", compl(x), compl(y)
  printf "0x%X, 0x%X, 0x%X\n", lshift(x,4), rshift(x,1), rshift(lshift(x,5),1)
}
