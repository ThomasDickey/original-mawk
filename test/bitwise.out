0x1, 0x13, 0x12
0xFFFFFFEE, 0xFFFFFFFC
0x110, 0x8, 0x110
